File: programmers/PRO77484.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int correctCnt = 0; // 일치횟수
        int zeroCnt = 0; // 0 의 개수
        
        for (int lotto : lottos) {
            if (lotto == 0) {
                zeroCnt++;
            } else if (Arrays.stream(win_nums).anyMatch(i -> i == lotto)) {
                correctCnt++;
            }
        }
        
        Map<Integer, Integer> map = new HashMap<>();
        for (int i=6; i > 0; i--) {
            map.put(i, 7-i);
        }
        map.put(0, 6);
        
        return new int[]{map.get(correctCnt + zeroCnt), map.get(correctCnt)};
    }
}

File: programmers/2021_Dev-Matching/PRO77484/PRO77484.py
def solution(lottos, win_nums):
    answer = []
    
    correct = 0
    zeroCount = 0
    
    for lotto in lottos:
        if lotto in win_nums:
            correct += 1
        elif lotto == 0:
            zeroCount += 1
            
    minLank = 6
    if correct > 0:
        minLank = 7 - correct
    
    maxCount = correct + zeroCount
    maxLank = 6
    if maxCount > 0:
        maxLank = 7 - maxCount
        
    answer.append(maxLank)
    answer.append(minLank)
    
    return answer